src/dune_rules/pkg_rules.ml

```diff
@@ -1937,6 +1938,23 @@ let build_rule context_name ~source_deps (pkg : Pkg.t) =
        ~directory_targets:[ pkg.write_paths.target_dir ]
 ;;
 
+let build_packages_of_context ctx_name =
+  let+ packages =
+    (* Fetching the package target implies that we will also fetch the extra sources. *)
+    Package_universe.lock_dir (Project_dependencies ctx_name)
+    >>| (fun lock_dir -> lock_dir.packages)
+    >>| Dune_lang.Package_name.Map.keys
+  in
+  let open Action_builder.O in
+  packages
+  |> List.map ~f:(fun pkg ->
+    Paths.make ~relative:Path.Build.relative (Project_dependencies ctx_name) pkg
+    |> Paths.target_dir
+    |> Path.build)
+  |> Action_builder.paths
+  >>> Action_builder.return @@ Action.Full.make (Action.Progn [])
```

Comment:
@rgrinberg To be able to build a `Action.Full.t` we used an empty `Progn`: what would be a better way to get a `Action.Full.t Action_builder.t` from our `unit Action_builder.t`?

Reply:
There is. Pushed a fix.